gabrielaEnache/src/main/java/oop/store/AbstractOrder.java

```diff
@@ -0,0 +1,106 @@
+package oop.store;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class AbstractOrder implements Order {
+    protected final long id;
+    protected final LocalDateTime data;
+    protected List<Item> items;
+
+    protected AbstractOrder(long id, List<Item> items) {
+        this.id = id;
+        this.data = LocalDateTime.now();
+        this.items = items;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public LocalDateTime getData() {
+        return data;
+    }
+
+    public List<Item> getItems() {
+        return items;
+    }
+
+    public void setItems(List<Item> items) {
+        this.items = items;
+    }
+
+    public void addItem(Item item) {
+        this.items.add(item);
+    }
+
+    public void addAllItems(List<Item> list) {
+        this.items.addAll(list);
+    }
+
+    public void removeItem(Item item) {
+        this.items.remove(item);
+    }
+
+    public void clearItemList() {
+        this.items.clear();
+    }
+
+    public void removeMultipleItems(List<Item> list) {
+        this.items.removeAll(list);
+    }
+
+    public static List<UserOrder> filterOrdersByItemType(List<UserOrder> listToSort, Type type) {
+        List<UserOrder> result = new ArrayList<>();
+        for (UserOrder userOrder : listToSort) {
+            if (userOrder.containsItemType(type))
+                result.add(userOrder);
+        }
+        return result;
+    }
+
+    public boolean containsItemType(Type type) {
+        for (Item item : this.getItems()) {
+            if (type.equals(item.getType()))
+                return true;
+        }
+        return false;
+    }
+
+    //Overloaded method
+    public float sum(String itemName) {
+        float result = 0;
+        for (Item item : items)
+            if (item.getTitle().equalsIgnoreCase(itemName))
+                result += item.getPrice();
+        return result;
+    }
+
+    //Overloaded method
+    public int count(String itemName) {
```

Comment:
Good approach, could you please overload this method by using an Item as method input parameter?

Reply:
Fixed